### ISSUE TITLE
Forward Direction in visualize_sub_traj.py

### ISSUE THREAD

**Issue by qiaogh97** [UCSB-AI/Aerial-Vision-and-Dialog-Navigation#10]
Dear authors,
         From the ./datasets/README.md, I know the meaning of "gt_path_corners". Therefore, I draw the corner[0] in a white circle and the corner[1] in a black circle using these lines of code.

    ```
    for i, corner in enumerate(pos_list):
        if i==(len(pos_list)-1):
            cv2.circle(im_resized, corner[0], color=(255, 255, 255), radius=16, thickness=-1)
            cv2.circle(im_resized, corner[1], color=(0, 0, 0), radius=16, thickness=-1)
    ```

This is the output picture.
    
<img width="484" alt="image" src="https://github.com/eric-ai-lab/Aerial-Vision-and-Dialog-Navigation/assets/20733440/74f5d04c-21e1-4bf5-9b1e-fdbd11dfd4cb">
   
   I think the white circle stands for "front left" and the black circle stands for "front right". I am confused because the Forward Direction arrow doesn't point to the midpoint of the white and black circles.

**Comment by UeFan** [comment]
Hi,

Thanks for reporting this. After careful check, there are two bugs causing this issue. Luckily none of them affect the training and evaluation results.

My answer to you earlier is correct that corner[0] is the front left corner and corner[1] is the front right.

The bugs are:
1. The function get_direction in datasets/visualize_sub_traj.py is applied to wrong target. The get_direction function can only work on GPS coordinates (in our code it should be [Latitude, Longitude]). However, this get_direction function was applied to the image coordinate ([column, row]) by mistake. I have now fixed this bug by replace the get_direction with a new function get_direction_for_img_coord.

2. The corners of the last view area are in the wrong order. You might noticed that the reversed white and black circles are only curred for certain sub-tractories. This is because the **final** view area in each trajectory in our dataset (not sub-trajectory) has a wrong corner order. Luckily, the corner[0] and corner[1] still represent the front direction, but just wrong in the left/right. So in general this does not affect the evaluation or training as we always use the center of view area by taking the mean of the four corners. I have updated the dataset.
